Automattic/jetpack
# Podcast Player: add colors setting to the block

This PR adds the colors setting feature to the Podcast Player block.
It adds three kinds of colors that will be applied according to the following criteria.

1) Primary color (darker grey used in episode title at top, in the darker colored icons in the player, and in the selected episode in list below). 
2) Secondary color (light grey used in podcast name, most player controls, description, and non-selected episodes in the list below, and open in new tab)
3) Background color

<img width="931" alt="Screen Shot 2020-03-25 at 6 55 11 PM" src="https://user-images.githubusercontent.com/77539/77589600-bb1e5d80-6eca-11ea-9b83-06407f11be2e.png">

![image](https://user-images.githubusercontent.com/77539/77704586-2c7a1100-6f9c-11ea-830d-a54d6d70e985.png)

### Technical implementation

In terms of technical implementation, the main goal is to keep the universality of the block as much as possible. It means that colors are handled by the same code for both sides of the app: editor-canvas and front-end.
For this purpose, I've moved the attributes into the `attributes` property of the `<PodcastPlayer />` component:

Inspecting properties in the **editor-canvas**:
![image](https://user-images.githubusercontent.com/77539/77605317-5b887800-6ef3-11ea-9d0f-daf6a6d9d10b.png)

and in the **front-end**:
![image](https://user-images.githubusercontent.com/77539/77605351-6fcc7500-6ef3-11ea-9428-9de03d283a59.png)

Fixes https://github.com/Automattic/jetpack/issues/14940

#### Changes proposed in this Pull Request:

*

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here.

#### Testing instructions:

* Apply the patch.
* Set colors through of the inspector controls in the sidebar. Confirm that colors are stored and shown in the front-end.

#### Proposed changelog entry for your changes:

* Add `primary`, `secondary` and `background` colors in the inspector controls.
* Move attributes into its own field.
* Apply colors on both sides via client-side scrips.
* Implement contrast checker between secondary and background color.
